STABILITY.md

```diff
@@ -0,0 +1,123 @@
+    │    └───────────────────────────────────┘    │
+    └─────────────────────────────────────────────┘
+ ```
+
+### Tier 3: pre-release crates
+
+Pre-release crates is marked with major version number zero, for example `0.1.4` and lives in stability tier 3 _(->ST3)_.
+
+It's acceptable to let each breaking change be followed by a minor version release.
+Dependent crates are advised to use `=<version>` version requirements to avoid automatic updates of patch levels which
+may be breaking if they also depend on other `plumbing crates`.
+
+### Tier 2: released plumbing crates
+
+Released plumbing crates are marked with major version number 1 or above, for example `1.2.4` and live in stability tier 2 _(->ST2)_.
+
+Breaking changes are collected and may be released no more often than every 4 weeks by incrementing the major version number.
+
+For example, `git-odb` and `git-ref` both have breaking changes, where `git-odb`'s change is on August 1st and `git-ref`'s dependent change
+is on September 10th. `git-odb`'s breaking change can be released earliest on September 1st, whereas `git-ref` can be released earliest at October 10th.
+
+If there are additional breaking changes without a release, these push back the earliest release date accordingly.
+
+### Tier 1: released apps and application crates
+
+Released apps and application crates are marked with major version number 1 or above, like `2.3.0+21.06` and live in tier 1 _(->ST1)_, 
+with the build identifiers for year (`21`) and and month `06` appended, based on the actual release year and month.
+
+Breaking changes are collected and may be released no more often than every 6 months by incrementing the major version number. If there are additional breaking changes,
+these push bac the release date so that they can be tested at least for 3 months. For example, a breaking change happens in January 01, and another breaking change in February 15.
+The earliest release date is July 1st. Had the second breaking change happened in April 01, the release date would have to be pushed to August 1st.
+
+Intermediate pre-releases may be created at most every 4 weeks by appending `-alpha.X` where `X` is the sequential release number. These should help testing
+breaking changes or new features without forcing the use of `git` sources for dependencies in cargo manifests. Pre-release releases must pin all the pre-release
+crates they depend on to prevent automatic upgrades.
+
+Minor version updates for new features can be released when needed assuming there are no other breaking changes, updating the build identifiers for year and month accordingly.
+
+## The _Minimal Stable Rust Version_ (->MSRV)
+
+It is as low as it can be for the crate in questions. While there is no non-pre-release of a tier 1 crate, the MSRV is automatically assumed to be the latest stable version.
+
+Increasing the MSRV is not considered a breaking change and doesn't warrant a major version bump itself.
```

Comment:
Is this convention? My intuition of semver would be that bumping MSRV would be considered backwards incompatible, but I suppose this isn't part of the API, so semver doesn't strictly apply here.

Should we make it at least a minor version bump? And/or suggest that we will attempt (but not guarantee) to only change msvp at minor or major versions?

Reply:
> Is this convention? My intuition of semver would be that bumping MSRV would be considered backwards incompatible, but I suppose this isn't part of the API, so semver doesn't strictly apply here.

That was also my first intuition, strictly speaking the MSRV (as well as the edition) affect stability and can break existing users. However, from what I reckon in the MSRV implementation for cargo nightly, cargo will be able to select only crates that have a matching msrv. So it won't select a patch level that would potentially break you.

> Should we make it at least a minor version bump? 

Whether minor or patch, it's the same effect on downstream crates as without cargo support, they will automatically update.

> And/or suggest that we will attempt (but not guarantee) to only change msvp at minor or major versions?

Something like this I had written there before but removed it for the sake of simplicity and the preference for "let's wait for actual demand before complicating our life".

After re-reading this paragraph I realised it was a little 'split' with portions of the old text and the new one. It has now made clear that the MSRV is the latest stable unless there is demand for setting it. Hence this is likely to be revisited at some point.